storage_v2/spanstore/reader.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package spanstore
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	spanstore_v1 "github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// ErrTraceNotFound is returned by Reader's GetTrace if no data is found for given trace ID.
+var ErrTraceNotFound = spanstore_v1.ErrTraceNotFound
+
+// Reader finds and loads traces and other data from storage.
+type Reader interface {
+	// GetTrace retrieves the trace with a given id.
+	//
+	// If no spans are stored for this trace, it returns ErrTraceNotFound.
+	GetTrace(ctx context.Context, traceID pcommon.TraceID) (*ptrace.Traces, error)
+
+	// GetServices returns all service names known to the backend from spans
+	// within its retention period.
+	GetServices(ctx context.Context) ([]string, error)
+
+	// GetOperations returns all operation names for a given service
+	// known to the backend from spans within its retention period.
+	GetOperations(ctx context.Context, query OperationQueryParameters) ([]Operation, error)
+
+	// FindTraces returns all traces matching query parameters. There's currently
+	// an implementation-dependent ambiguity whether all query filters (such as
+	// multiple tags) must apply to the same span within a trace, or can be satisfied
+	// by different spans.
+	//
+	// If no matching traces are found, the function returns (nil, nil).
+	FindTraces(ctx context.Context, query *TraceQueryParameters) ([]*ptrace.Traces, error)
```

Comment:
```suggestion
	FindTraces(ctx context.Context, query *TraceQueryParameters) ([]ptrace.Traces, error)
```
if you return 30 traces, `[]*...` requires 30 heap allocations for no practical gains.